### ISSUE TITLE
jinja[invalid] - false positive detection with json_query expression

### ISSUE THREAD

**Issue by jantari** [ansible/ansible-lint#2665]
##### Summary

The `jinja[invalid]` rule reports a false positive (supposedly invalid Jinja epxression).

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
$ ansible --version
ansible [core 2.12.9]
  config file = None
  configured module search path = ['/home/jantari/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/jantari/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/jantari/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/jantari/.local/bin/ansible
  python version = 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0]
  jinja version = 3.0.3
  libyaml = True

$ ansible-lint --version
ansible-lint 6.8.6 using ansible 2.12.9
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

I can reproduce the bug locally with Ubuntu 22.04.1 LTS, but we first discovered it on our GitLab CI pipelines that run Alpine Linux.

##### STEPS TO REPRODUCE

Construct the following directory structure:

```
.
├── requirements.yml
├── site.yml
└── tasks
    └── tasks.yml
```

With the file contents:

**requirements.yml**

```yaml
---

collections:
  - name: fortinet.fortios
    version: 2.1.7
```

**site.yml**

```yaml
---

- hosts: all
  gather_facts: false
  vars:
    vdom: root
    fortios_access_token: fake-token
  connection: httpapi
  collections:
    - fortinet.fortios
  tasks:
    - name: include
      include_tasks: tasks/tasks.yml
```

**tasks/tasks.yml**

```yaml
---

- name: Get all available physical interfaces
  fortinet.fortios.fortios_monitor_fact:
    vdom: "{{ vdom }}"
    access_token: "{{ fortios_access_token | default(omit) }}"
    selector: "system_available-interfaces"
    filters:
      - type==physical
  register: _available_interfaces
  check_mode: false

- name: Save list of all configurable interfaces # noqa jinja[spacing]
  ansible.builtin.set_fact:
    _configurable_interfaces: >
      {{ _available_interfaces.meta.results | json_query("[?!mac_address || mac_address != '00:00:00:00:00:00'].name") | list | sort }}

- name: Print all physical interfaces with non-null mac address
  ansible.builtin.debug:
    var: _configurable_interfaces
```

##### Desired Behavior

Running `ansible-lint -c /dev/null` in the above repro directory should return without any violations.
The playbook / tasks execute without error and as intended when run, the Jinja expressions are not invalid.

##### Actual Behavior

```
$ ansible-lint -c /dev/null
WARNING  Listing 1 violation(s) that are fatal
jinja[invalid]: Unexpected templating type error occurred on ({{ _available_interfaces.meta.results | json_query("[?!mac_address || mac_address != '00:00:00:00:00:00'].name") | list | sort }}
): 'NoneType' object is not iterable
tasks/tasks.yml:13 Task/Handler: Save list of all configurable interfaces

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - jinja[invalid]  # Rule that looks inside jinja2 templates.

              Rule Violation Summary
 count tag            profile rule associated tags
     1 jinja[invalid] basic   formatting

Failed after min profile: 1 failure(s), 0 warning(s) on 3 files.
```

##### Additional Context

The reproducible example of this error is an excerpt from a larger project of production code that contains many expressions where we gather facts from FortiOS firewall devices and filter them in similar ways, for example:

```jinja
{{ _ca_certificates.meta.results | selectattr( ... ) ... }}
```

or:

```jinja
{{ _current_sourcenat_rules.meta.results[ansible_loop.index0].policyid ... }}
```

and none of those are considered problematic by ansible-lint, it's only that one task with the specific `json_query()` expression.

Thanks!

**Comment by jantari** [comment]
This issue still exists in the current `ansible-lint 6.9.0 using ansible 2.12.9`
